cnf-tests/testsuites/e2esuite/n3000/n3000.go

```diff
@@ -0,0 +1,71 @@
+package n3000
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	fpgav1 "github.com/open-ness/openshift-operator/N3000/api/v1"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	n3000DaemonNamespace     = "n3000-operators-resources"
+	TestNamespace            = "n3000-testing"
+	n3000DeploymentName      = "n3000-controller-manager"
+	daemonsetDriverName      = "fpga-driver-daemonset"
+	daemonsetTelemetryName   = "fpga-driver-daemonset"
+	daemonsetN3000DaemonName = "n3000-daemonset"
+)
+
+var _ = Describe("Intel n3000: ", func() {
+	var n3000NodeList = &fpgav1.N3000NodeList{}
+
+	BeforeEach(func() {
+		if !n3000Exists(n3000NodeList) {
+			Skip("Cluster doesn't have intel fpga n3000 nic")
+		}
+	})
+
+	Context("N3000 operator validation: ", func() {
+		// 39355
+		It("should have at least 1 intel n3000 nic", func() {
+			Expect(n3000NodeList.Items).NotTo(BeNil(),"No n3000nodes are configured")
+
+			for _, n3000Node := range n3000NodeList.Items {
+				Expect(len(n3000Node.Status.Fortville[0].NICs)).To(BeNumerically(">", 0), fmt.Sprintf("No Intel fpga n3000 NICs are presented on the n3000node %s", n3000Node.Name))
+			}
+		})
+		// 39356
+		It("should have a ready deployment n3000-controller-manager", func() {
+			deployment, err := client.Client.Deployments(n3000DaemonNamespace).Get(context.Background(), n3000DeploymentName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(deployment.Status.ReadyReplicas).To(Equal(int32(1)), "Deployment n3000-controller-manager is not ready")
+		})
+		// 39314
+		It("should have all the required n3000 operator daemonsets", func() {
+			numberN3000Nodes := int32(len(n3000NodeList.Items))
+			daemonsetDriver, err := client.Client.DaemonSets(n3000DaemonNamespace).Get(context.Background(), daemonsetDriverName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			daemonsetTelemetry, err := client.Client.DaemonSets(n3000DaemonNamespace).Get(context.Background(), daemonsetTelemetryName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			daemonsetN3000Daemon, err := client.Client.DaemonSets(n3000DaemonNamespace).Get(context.Background(), daemonsetN3000DaemonName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(daemonsetDriver.Status.NumberReady).To(Equal(numberN3000Nodes), fmt.Sprintf("Daemonset %s is not ready", daemonsetDriverName))
+			Expect(daemonsetTelemetry.Status.NumberReady).To(Equal(numberN3000Nodes), fmt.Sprintf("Daemonset %s is not ready", daemonsetTelemetryName))
+			Expect(daemonsetN3000Daemon.Status.NumberReady).To(Equal(numberN3000Nodes), fmt.Sprintf("Daemonset %s is not ready", daemonsetN3000DaemonName))
+		})
+	})
+})
+
+func n3000Exists(n3000NodeList *fpgav1.N3000NodeList) bool{
+	err := client.Client.List(context.Background(), n3000NodeList)
```

Comment:
Just to double check, does this return an error or fill the list as empty list?
In case it returns an error, can you specifically check that it is a not found error? Otherwise it would eat unwanted errors.